beacon-chain/rpc/eth/rewards/handlers.go

```diff
@@ -166,22 +175,285 @@ func (s *Server) BlockRewards(w http.ResponseWriter, r *http.Request) {
 	network.WriteJson(w, response)
 }
 
+// AttestationRewards retrieves attestation reward info for validators specified by array of public keys or validator index.
+// If no array is provided, return reward info for every validator.
+// TODO: Inclusion delay
+func (s *Server) AttestationRewards(w http.ResponseWriter, r *http.Request) {
+	st, ok := s.attRewardsState(w, r)
+	if !ok {
+		return
+	}
+	bal, vals, valIndices, ok := attRewardsBalancesAndVals(w, r, st)
+	if !ok {
+		return
+	}
+	idealRewards, ok := idealAttRewards(w, st, bal)
+	if !ok {
+		return
+	}
+	totalRewards, ok := totalAttRewards(w, st, bal, vals, valIndices)
+	if !ok {
+		return
+	}
+
+	optimistic, err := s.OptimisticModeFetcher.IsOptimistic(r.Context())
+	if err != nil {
+		errJson := &network.DefaultErrorJson{
+			Message: "Could not get optimistic mode info: " + err.Error(),
+			Code:    http.StatusInternalServerError,
+		}
+		network.WriteError(w, errJson)
+		return
+	}
+	blkRoot, err := st.LatestBlockHeader().HashTreeRoot()
+	if err != nil {
+		errJson := &network.DefaultErrorJson{
+			Message: "Could not get block root: " + err.Error(),
+			Code:    http.StatusInternalServerError,
+		}
+		network.WriteError(w, errJson)
+		return
+	}
+
+	resp := &AttestationRewardsResponse{
+		Data: AttestationRewards{
+			IdealRewards: idealRewards,
+			TotalRewards: totalRewards,
+		},
+		ExecutionOptimistic: optimistic,
+		Finalized:           s.FinalizationFetcher.IsFinalized(r.Context(), blkRoot),
+	}
+	network.WriteJson(w, resp)
+}
+
+func (s *Server) attRewardsState(w http.ResponseWriter, r *http.Request) (state.BeaconState, bool) {
+	segments := strings.Split(r.URL.Path, "/")
+	requestedEpoch, err := strconv.ParseUint(segments[len(segments)-1], 10, 64)
+	if err != nil {
+		errJson := &network.DefaultErrorJson{
+			Message: "Could not decode epoch: " + err.Error(),
+			Code:    http.StatusBadRequest,
+		}
+		network.WriteError(w, errJson)
+		return nil, false
+	}
+	if primitives.Epoch(requestedEpoch) < params.BeaconConfig().AltairForkEpoch {
+		errJson := &network.DefaultErrorJson{
+			Message: "Attestation rewards are not supported for Phase 0",
+			Code:    http.StatusNotFound,
+		}
+		network.WriteError(w, errJson)
+		return nil, false
+	}
+	currentEpoch := uint64(slots.ToEpoch(s.TimeFetcher.CurrentSlot()))
+	if requestedEpoch+1 >= currentEpoch {
+		errJson := &network.DefaultErrorJson{
+			Code:    http.StatusNotFound,
+			Message: "Attestation rewards are available after two epoch transitions to ensure all attestations have a chance of inclusion",
```

Comment:
Not sure if it's a must but should message be same with the [API spec](https://ethereum.github.io/beacon-APIs/#/Rewards/getAttestationsRewards)?
ex) "Epoch not known or required data not available" for 404